### ISSUE TITLE
sphtfunc.pixwin returns array of size greater than default lmax=3*nside

### ISSUE THREAD

**Issue by msyriac** [healpy/healpy#536]
This probably just needs clarification or an API extension, but see

```
>>> import healpy as hp
>>> import numpy as np
>>> nside = 128
>>> npix = hp.nside2npix(nside)
>>> hmap = np.random.random((npix,))
>>> alm = hp.map2alm(hmap)
>>> cl = hp.alm2cl(alm)
>>> cl.shape
(384,)
>>> hp.pixwin(nside).shape
(513,)
```

i.e., map2alm defaults to lmax=3 * nside, while pixwin presumably defaults to lmax=4 * nside. Is that correct? If so, it would help to clarify this in the docs, and also possibly extend the pixwin API so that it returns the pixel window up to a desired lmax.

**Comment by DanielLenz** [comment]
Thanks for raising this issue! I support the idea of having an `lmax` keyword, default should be either 3*nside or whatever lmax the pixwins are calculated up to.

If there are no objections, I'll add a PR for this. @zonca

**Comment by mreineck** [comment]
I've probably mentioned this in the past; if so, sorry for the repetition...
Giving a default lmax for `map2alm` is a bad idea. Choosing a good value requires knowledge about the map to be analyzed. In most cases I'd recommend `lmax=2*nside`, values should only ever be higher if the user is sure that the map is properly band-limited (which is virtually never the case, unless you just created the map using `alm2map`). Values above `3*nside` definitely make no sense; I have no idea why the pixel window values go further up.

**Comment by zonca** [comment]
@DanielLenz I think it should be the same as `map2alm`

@mreineck do you have a reference that explains how you are supposed to choose `lmax`? It would be useful to link it or give a summary in the `map2alm` docs.

**Comment by mreineck** [comment]
The most thorough user guide for harmonic analysis on Healpix maps is probably given in the `anafast` documentation from the Fortran package: https://healpix.sourceforge.io/html/fac_anafast.htm

**Comment by DanielLenz** [comment]
@zonca Okay, I'll just use the following:
```
if lmax is None:
        lmax_ = 3 * nside - 1
    else:
        lmax_ = lmax

return pw_temp[:lmax_]
```

**Comment by zonca** [comment]
@DanielLenz please crosscheck but I think it might be `[:lmax+1]`

**Comment by zonca** [comment]
closed by #544
